scipy/stats/_mannwhitneyu.py

```diff
@@ -471,14 +456,14 @@ def mannwhitneyu(x, y, use_continuity=True, alternative="two-sided",
 
     n1, n2 = x.shape[-1], y.shape[-1]
 
-    if method == "auto":
-        method = _mwu_choose_method(n1, n2, xy, method)
-
     # Follows [2]
-    ranks = stats.rankdata(xy, axis=-1)  # method 2, step 1
-    R1 = ranks[..., :n1].sum(axis=-1)    # method 2, step 2
-    U1 = R1 - n1*(n1+1)/2                # method 2, step 3
-    U2 = n1 * n2 - U1                    # as U1 + U2 = n1 * n2
+    ranks, t = _rankdata(xy, 'average', return_ties=True)  # method 2, step 1
+    R1 = ranks[..., :n1].sum(axis=-1)                      # method 2, step 2
+    U1 = R1 - n1*(n1+1)/2                                  # method 2, step 3
+    U2 = n1 * n2 - U1                                      # as U1 + U2 = n1 * n2
+
+    if method == "auto":
+        method = _mwu_choose_method(n1, n2, np.any(t > 1), method)
```

Comment:
I think this two lines 465 and 466 can be moved to before line 475 for better readability because the `method` is used from line 475.

Reply:
Ok.